mlflow/tracing/provider.py

```diff
@@ -154,4 +175,4 @@ def _is_enabled() -> bool:
         if isinstance(tracer, trace.ProxyTracer):
             tracer = tracer._tracer
 
-        return not isinstance(tracer, trace.NoOpTracer)
+        return not (_TRACER_PROVIDER_INITIALIZED._done and isinstance(tracer, trace.NoOpTracer))
```

Comment:
note: The original condition was faulty indeed, we should also treat the default state (before any tracer_provider is set up) as `enabled` - because tracing is on by default.